src/dom.js

```diff
@@ -1215,7 +1215,24 @@ export const getCurrentScriptUID : GetCurrentScriptUID = memoize(() => {
         return uid;
     }
 
-    uid = uniqueID();
+    if (script.src) {
+
+        let stringToHash = script.src;
+
+        Object.keys(script.dataset).forEach(key => {
+            if (key === 'uid') {
+                return;
+            }
+            stringToHash += script.dataset[key];
```

Comment:
I think we should include the key name also. Can we do a `JSON.stringify({ url, attributes })` where attributes is a mapping of key->value?

Reply:
Whoops I forgot about the attribute keys. Good suggestion.

Reply:
I have implemented the suggested approach.